zipkin-lens/src/components/UiConfig/UiConfig.test.jsx

```diff
@@ -13,65 +13,60 @@
  */
 import { render, screen } from '@testing-library/react';
 import fetchMock from 'fetch-mock';
-import { afterEach, beforeEach, it, describe, expect } from 'vitest';
+import { afterEach, it, describe, expect, vi } from 'vitest';
 import React, { Suspense } from 'react';
 
-import {
-  defaultConfig,
-  UiConfigConsumer,
-  UiConfig as RawUIConfig,
-} from './UiConfig';
-import { UI_CONFIG } from '../../constants/api';
-
 afterEach(() => {
   fetchMock.restore();
+  vi.resetModules();
 });
 
-beforeEach(() => {
-  // We fetch the resource on module initialization for performance, but want to do that in every
-  // test.
-});
-
-const UiConfig = () => {
-  return (
+const renderUiConfig = async () => {
+  const { UiConfigConsumer, UiConfig } = await import('./UiConfig');
+  render(
     <Suspense fallback="Suspended">
-      <RawUIConfig>
+      <UiConfig>
         <UiConfigConsumer>
           {(value) => <div>{JSON.stringify(value)}</div>}
         </UiConfigConsumer>
-      </RawUIConfig>
-    </Suspense>
+      </UiConfig>
+    </Suspense>,
   );
 };
 
 describe('<UiConfig />', () => {
   it('fetches config and suspends', async () => {
     const configPromise = new Promise(() => undefined);
+    const { UI_CONFIG } = await import('../../constants/api');
     fetchMock.once(UI_CONFIG, configPromise, { overwriteRoutes: true });
 
-    render(<UiConfig />);
+    await renderUiConfig();
     expect(screen.getAllByText('Suspended')).length(1);
 
     fetchMock.called(UI_CONFIG);
   });
 
-  it('provides config when resolved', async () => {
+  it('provides config when resolved', (context) => async () => {
```

Comment:
since this file was renamed in the last commit, none of the tests ran. I was able to get the former to run and pass, but it is a bit hacky as I really have no idea what I'm doing.

@anuraaga this is on a shared branch.. I think your old test was trying to show that the UI can work while the config.json request hasn't completed yet? I tried for over an hour to try to get this to work, so am not bothering you randomly.. could really use a hand (even you can paste) if you can help get this working.

Reply:
https://github.com/openzipkin/zipkin/issues/3724 for follow-up